hibike/hibike_message.py

```diff
@@ -405,17 +415,18 @@ def parse_bytes(msg_bytes):
     return HibikeMessage(message_id, payload)
 
 
-def blocking_read_generator(serial_conn, stop_event=None):
+def blocking_read_generator(serial_conn, stop_event=threading.Event()):
     """
-    Yield packets from SERIAL_CONN, stopping if STOP_EVENT exists
-    and is set.
+    Yield packets from SERIAL_CONN, stopping if STOP_EVENT is set.
     """
     zero_byte = bytes([0])
     packets_buffer = bytearray()
-    while stop_event is None or not stop_event.is_set():
-
+    # Switch to nonblocking mode so that we don't get stuck reading
+    old_timeout = serial_conn.timeout
+    serial_conn.timeout = 0
+    while not stop_event.is_set():
         # Wait for a 0 byte to appear
-        while packets_buffer.find(zero_byte) == -1:
+        while packets_buffer.find(zero_byte) == -1 and not stop_event.is_set():
```

Comment:
I'm sorry, I don't quite understand what you mean by that. Which part am I changing into an `if`?